src/interface/mesh_data.hpp

```diff
@@ -289,50 +289,211 @@ class MeshData {
+                                     PackIndexMap &map, vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(&map, &key, names_or_flags);
+  }
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(nullptr, &key, names_or_flags);
   }
-
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, PackIndexMap &map,
-                            std::vector<std::string> &key) {
-    return PackVariablesImpl(&map, &key, std::forward<Args>(args)...);
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr, names_or_flags);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, PackIndexMap &map) {
-    return PackVariablesImpl(&map, nullptr, std::forward<Args>(args)...);
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags) {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr, names_or_flags);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, std::vector<std::string> &key) {
-    return PackVariablesImpl(nullptr, &key, std::forward<Args>(args)...);
+  // only sparse ids
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     PackIndexMap &map, vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(&map, &key, sparse_ids);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args) {
-    return PackVariablesImpl(nullptr, nullptr, std::forward<Args>(args)...);
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(nullptr, &key, sparse_ids);
+  }
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr, sparse_ids);
+  }
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids) {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr, sparse_ids);
+  }
+  // No nothing
+  const auto &PackVariablesAndFluxes(PackIndexMap &map, vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(&map, &key);
+  }
+  const auto &PackVariablesAndFluxes(vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(nullptr, &key);
+  }
+  const auto &PackVariablesAndFluxes(PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr);
+  }
+  const auto &PackVariablesAndFluxes() {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr);
+  }
```

Comment:
What does this return with no arguments? (For some reason I can't find the two-argument version of `PackVariablesAndFluxesImpl` to look at the code itself)

Reply:
It should pack all variables and all fluxes.